nft/ethereum_payment_obligation.go

```diff
@@ -204,6 +204,17 @@ func (s *ethereumPaymentObligation) minter(ctx context.Context, tokenID TokenID,
 			return
 		}
 
+		// Check if tokenID exists in registry and owner is deposit address
+		owner, err := s.OwnerOf(req.RegistryAddress, tokenID[:])
```

Comment:
Just out of curiosity.
What happens if the user passes an invalid address as depositAddress?
Would it fail?

Reply:
good question, I think there must be a test case in truffle project that should verify that.

Reply:
What does invalid mean? There is a check for a valid ETH address.

Reply:
Valid means it passes a checksum test

Reply:
There are no checksum tests for addresses, the checksum is only in capitalization which is not really enforced outside of user input. Theoretically any 20 byte value is correct.

Reply:
Yes, but you can also enforce the checksum to have some extra validation. Web3 does that for example